wcomponents-test-lib/src/main/java/com/github/bordertech/wcomponents/test/selenium/by/ByButtonText.java

```diff
@@ -0,0 +1,85 @@
+package com.github.bordertech.wcomponents.test.selenium.by;
+
+import java.util.List;
+import org.openqa.selenium.By;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.internal.FindsByXPath;
+
+/**
+ * Finds a button by the text value on the button.
+ */
+public class ByButtonText extends By {
```

Comment:
A suggestion (rather than a requirement). We know that `findElementsByXPath` is slow in some browsers (though the leative search will help) and that byCssSelector is often significantly quicker. Can we demonstrate that this would be faster than getting a list of buttons (either by tag name or by css selector) and then iterating through them to find the button(s) with the required text. I have a feeling that for UIs with a lot of buttons the xpath selector method would be faster than testing the text content of each button but the tagname or CSS selector method may be quicker in UIs with few buttons.

Can you provide for both?